src/rust/catmem/mod.rs

```diff
@@ -80,11 +54,28 @@ const RING_BUFFER_CAPACITY: usize = 65536;
 type Operation = dyn Future<Output = (QDesc, OperationResult)>;
 type OperationTask = TaskWithResult<(QDesc, OperationResult)>;
 
+#[derive(Clone)]
+/// Operation Result
+pub enum OperationResult {
+    Push,
+    Pop(DemiBuffer),
+    Close,
+    Failed(Fail),
+}
+
+/// Whether this queue is open for push or pop (i.e., producer or consumer).
+pub enum QMode {
```

Comment:
Can't be this Encoded in QueueType, as we have for UdpSocket and TcpSocket?

Reply:
Yes, I think we could but generally we do not use QueueType in arguments to functions when creating the queue so I wasn't sure what to do.

Reply:
Should we add that as an argument to make the interface nicer? What is your opinion on this?

Reply:
As we discussed, this is be fixed on a different PR.